### PR TITLE
liboqs: 0.8.0 -> 0.10.0, migrate to by name

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- liboqs: `0.8.0` -> `0.10.0`
- migrate to by name

I faced the issue described in https://github.com/NixOS/nixpkgs/issues/144170 and fixed it by reverting [this commit](https://github.com/open-quantum-safe/liboqs/commit/f0e6b8646c5eae0e8052d029079ed3efa498f220) from upstream which change logic of `liboqs.pc` generation.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/liboqs/package.nix [hunk 1]
```diff
@@ -3,24 +3,35 @@
...
+    # causing malformed path with double slashes.
+    (fetchpatch {
+      url = "https://github.com/open-quantum-safe/liboqs/commit/f0e6b8646c5eae0e8052d029079ed3efa498f220.patch";
+      hash = "sha256-tDfWzcDnFGikzq2ADEWiUgcUt1NSLWQ9/HVWA3rKuzc=";
+      revert = true;
+    })
+  ];
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [ cmake ninja ];
```
Comment (NickCao) [inline pkgs/by-name/li/liboqs/package.nix:34]: Why ninja?

Comment (Sigmanificient) [reply pkgs/by-name/li/liboqs/package.nix:34]: The README showcase build with ninja: https://github.com/open-quantum-safe/liboqs?tab=readme-ov-file#linux-and-mac

Comment (NickCao) [reply pkgs/by-name/li/liboqs/package.nix:34]: The use of ninja is only beneficial for incremental builds. As long as gnumake works I'd prefer that, to reduce build closure size.

File: pkgs/by-name/li/liboqs/package.nix [hunk 2]
```diff
@@ -30,6 +40,7 @@ stdenv.mkDerivation rec {
   ];
 
   dontFixCmake = true; # fix CMake file will give an error
+  outputs = [ "out" "dev" ];
```
Comment (xzfc) [inline pkgs/by-name/li/liboqs/package.nix:43]: Nit: add a newline.
```suggestion

  outputs = [ "out" "dev" ];
```

Comment (Sigmanificient) [reply pkgs/by-name/li/liboqs/package.nix:43]: Can you explain why tho?

File: pkgs/by-name/li/liboqs/package.nix [hunk 3]
```diff
@@ -4,21 +4,31 @@
...
+  version = "0.10.0";
 
   src = fetchFromGitHub {
     owner = "open-quantum-safe";
-    repo = pname;
-    rev = version;
-    sha256 = "sha256-h3mXoGRYgPg0wKQ1u6uFP7wlEUMQd5uIBt4Hr7vjNtA=";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    sha256 = "sha256-BFDa5NUr02lFPcT4Hnb2rjGAi+2cXvh1SHLfqX/zLlI=";
```
Comment (drupol) [inline pkgs/by-name/li/liboqs/package.nix:19]:
```suggestion
    hash = "sha256-BFDa5NUr02lFPcT4Hnb2rjGAi+2cXvh1SHLfqX/zLlI=";
```

File: pkgs/by-name/li/liboqs/package.nix [hunk 4]
```diff
@@ -4,21 +4,31 @@
...
   pname = "liboqs";
-  version = "0.8.0";
+  version = "0.10.0";
 
   src = fetchFromGitHub {
     owner = "open-quantum-safe";
-    repo = pname;
-    rev = version;
-    sha256 = "sha256-h3mXoGRYgPg0wKQ1u6uFP7wlEUMQd5uIBt4Hr7vjNtA=";
+    repo = finalAttrs.pname;
```
Comment (drupol) [inline pkgs/by-name/li/liboqs/package.nix:17]: Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/liboqs/package.nix (hunk 4 above)
Comment (Sigmanificient) [reply pkgs/by-name/li/liboqs/package.nix:17]: Well, so old nix code moved

**Review by drupol — APPROVED** [review]
Thank you.

**Pull request merged**